tests/taskcluster/bin/raptor_graph

```diff
@@ -29,48 +37,115 @@ function params() {
+provided HEAD_REV; else it will be BASE_REV -1 commit
+*/
+
 var config = {
-  task1: {
+  baseTask1: {
     definition: 'raptor_launch',
     params: {
-      provisionerId: 'aws-provisioner-v1',
-      workerType: 'gaia',
-      raptorRuns: 15,
-      raptorApps: 'settings',
+      raptorSuiteName: 'Settings Launch Test (on base)',
       gaiaRev: process.env.GITHUB_BASE_REV,
-      raptorTimeout: 600000,
-      raptorBootSleep: 180,
+      raptorRuns: 30,
+      raptorApps: 'settings',
       treeherderSym: 'se1'
     }
   },
-  task2: {
+  patchTask1: {
     definition: 'raptor_launch',
     params: {
-      provisionerId: 'aws-provisioner-v1',
-      workerType: 'gaia',
-      raptorRuns: 15,
-      raptorApps: 'settings',
+      raptorSuiteName: 'Settings Launch Test (on base + pr)',
       gaiaRev: process.env.GITHUB_HEAD_REV,
-      raptorTimeout: 600000,
-      raptorBootSleep: 180,
+      raptorRuns: 30,
+      raptorApps: 'settings',
       treeherderSym: 'se2'
     }
   },
-  task3: {
+  resultTask: {
     definition: 'raptor_result',
+    requires: ['baseTask1', 'patchTask1'],
     params: {
-      completedBranchTask: '',
-      completedPatchTask: '',
-      provisionerId: 'aws-provisioner-v1',
-      workerType: 'gaia',
+      raptorSuiteName: 'Settings Results',
       treeherderSym: 'se'
     }
   }
 };
 
+/**
+ * Execute the built command in a child process
+ * @returns {Promise}
+ */
+var do_command = function(cmd) {
+  return new Promise(function(resolve, reject) {
+    exec(cmd, function(err, stdout, stderr) {
+      if (err) {
+        return reject(err, stderr);
+      }
+      resolve(stdout.trim());
+    });
+  });
+};
+
+/**
+ * Resolve when latest emulator build url has been retrieved
+ * @returns {Promise}
+ */
+var getEmulator = function() {
+  return new Promise(function(resolve, reject) {
+
+    // Get taskId for the last successful emulator-ics build task
+    var taskId = 99;
+    var ns = "buildbot.branches.mozilla-central.emulator-ics";
+
+    // Find task in namespace
+    var index = new taskcluster.Index(TC_CLIENT_OPTS);
+    index.findTask(ns)
+      .then(function(foundTask) {
+        // Url for build is
+        // https://queue.taskcluster.net/v1/task/{taskId}/artifacts/{artifact}
+        emulatorBuildURL = url.format({
+          protocol: 'https',
+          host: 'queue.taskcluster.net',
+          pathname: '/v1/task/' + foundTask.taskId + '/artifacts/public/build/emulator.tar.gz'
+        });
+        resolve();
+      });
+  });
+};
+
+/**
+ * Resolve when patchTask gaia rev has been retrieved and set
+ * When not a PR, we want patchTask rev to be BASE_REV -1 commit
+ * @returns {Promise}
+ */
+var setpatchTaskRev = function() {
+  return new Promise(function(resolve, reject) {
+    if (process.env.GITHUB_PULL_REQUEST === '0') {
+      var cmd = "cd /home/rwood/gaia/ && git rev-parse HEAD~1";
+      do_command(cmd).then(function(baseMinusOne) {
```

Comment:
You might be able to just use child_process.execSync so you don't have to have extra code for a promise lying around.  I don't think in this task container anything else is being done at the same time so tying up the event loop for a brief moment won't halt anything.

Of course then you need a try/catch around this in case the command files.  Either way, just a thought